src/EnergyPlus/UnitarySystem.cc

```diff
@@ -12851,6 +12851,11 @@ namespace UnitarySystems {
                     }
                 }
 
+                // If the coil is temperature controlled, reset the desired outlet temperature so the temperature setpoint at the oulet node won't be used
+                if (DataLoopNode::SensedLoadFlagValue < 0) {
+                    DataLoopNode::Node(OutletNode).Temp = 0;
+                }
```

Comment:
I don't understand this logic. DataLoopNode::SensedLoadFlagValue = -999.0 so it is always < 0. Also, the component models should be responsible for setting a node condition not the parent so this code should be removed (unless there is an extremely good reason to do it). Can I assume that with this code the unit test you wrote still passes now?

Reply:
I originally was doing something similar in HeatingCoils.cc instead but ran into some issues with other unit tests so I switched to putting this in the UnitarySystem.cc since it occurred for unitary systems.  Based on your comment, I'm thinking now I should put what I had back in HeatingCoils.cc and try to address the other unit test issues.  This seems out of place here now.

Reply:
Certainly don't do this:

  if (DataLoopNode::SensedLoadFlagValue < 0)

Reply:
If you can briefly describe the issue you see in code it might help in the review process.